ocaml/stdlib/camlinternalOO.ml

```diff
@@ -271,7 +276,7 @@ let new_variable table name =
     index
 
 let to_array arr =
-  if arr = Obj.magic 0 then [||] else arr
+  if arr = Sys.opaque_identity (Obj.magic 0) then [||] else arr
```

Comment:
I am certainly out of my depth here,
but was wondering why `=` was used
(as opposed to `==`)?

Reply:
I don't know; I just followed the existing code!

Reply:
Oh, indeed, it was not intended as a comment
about the PR, just genuine curiosity.